pkg/resiliency/retry.go

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package resiliency
+
+import (
+	"errors"
+)
+
+// IsRetryable checks if an error is classified as retryable.
+// Returns true if so, false otherwise.
+func IsRetryable(e error) bool {
+	re := new(RetryableErr)
+
+	errs, ok := e.(interface {
+		Unwrap() []error
+	})
+	if !ok {
+		return errors.As(e, re)
+	}
+	for _, err := range errs.Unwrap() {
```

Comment:
I wonder if we need this at all for a first cut of the package. I'm intending towards a usage pattern where the developer is writing code that errors and they would use `NewRetryableErr` and this error would be returned to the caller which would presumably be responsible for retrying, and if not, then delegating further up the stack by re-wrapping the error again with `NewRetryableErr`.  Eventually, some calling code will kick off the retry process. 

In that pattern, we don't necessarily need `Unwrap` and `errors.As`, but rather simply `errors.Is`.

The reason I'm thinking this is that I can't really imagine a scenario where there's a retryable error thrown deep in the call stack and every layer above it doesn't consider the error retryable.

Posting this to see if it also aligns with others' thinking.

Reply:
Tx Chris! Good point!! This call is now worthless as behaviors were axed.